api/src/Util/EntityMap.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+namespace App\Util;
+
+use App\Entity\BaseEntity;
+
+class EntityMap {
+    private $map = [];
+
+    public function add(BaseEntity $prototype, BaseEntity $entity) {
+        $key = $prototype::class.'#'.$prototype->getId();
+        $this->map[$key] = $entity;
+    }
+
+    public function get(BaseEntity $prototype): BaseEntity {
+        $key = $prototype::class.'#'.$prototype->getId();
+        $keyExists = array_key_exists($key, $this->map);
+
+        return $keyExists ? $this->map[$key] : $prototype;
```

Comment:
Finde ich bisschen gefährlich, dass man hier nicht mitbekommt, wenn etwas in der EntityMap fehlt.

Reply:
Werden wir brauchen:
- Camp kopieren: Da werden kopierte Aktivitäten auf eine Kopierte Kategorie zeigen (in Map enthalten)
- Aktivität kopieren: Da wird kopierte Aktivität auf bestehende Kategorie zeigen. (nicht in Map enthalten)

Aktuell kopieren wir keine Aktivitäten - ich habe dieses Beispiel gewählt, weil es leicht verständlich ist.
Das existierende Beispiel aktuell sind MaterialItems (ContentNode & MaterialList)

Reply:
Das heisst wir brauchen diesen code momentan nicht (und er ist auch nicht getestet).
Ich würde vorschlagen, dass wir jetzt in diesem fall eine exception werfen, und in einem Unittest für EntityMap testen, dass die Exception geworfen wird.

Wenn wir das kopieren der acitivty implementieren, dann fügt man in ActivityDataPersister::onBeforeCreate die category, an die wir die activity hängen wollen, in die EntityMap ein.

Wenn wir dann eine activity von einem camp in ein anderes kopieren, muss der User angeben, was die neue category der activity sein soll. (Eventuell kann man mit der prototypeId magic machen, aber das wird nicht immer funktionieren).

Reply:
so lassen

Reply:
## Core Meeting Decision
So lassen